data/sea/54-zebra.h

```diff
@@ -206,22 +270,42 @@ void zebra_collect_state (zebra_config_t *cfg, zebra_state_t *state)
     free (state);
 }
 
+ierror_msg_t zebra_read_step (piano_t *piano, zebra_state_t *state, zebra_entity_t *entity)
+{
+    ifleet_t *fleet = state->fleet;
+    psv_collect_fleet (fleet);
+
+    ierror_loc_t  error_loc = psv_configure_fleet ((char*) entity->id_bytes, entity->id_length, piano, state->fleet);
+    if (error_loc) return ierror_loc_pretty (error_loc, 0);
+
+    ierror_msg_t error = zebra_read_entity (piano, state, entity);
+    if (error) return error;
+
+    /* keep going if we have not finished reading all facts for this entity */
+    for (int64_t i = 0; i != state->attribute_count; ++i) {
+        if (state->entity_fact_offset[i] != state->entity_fact_count[i]) {
+            return zebra_read_step (piano, state, entity);
+        }
+        state->entity_fact_count [i] = 0;
+        state->entity_fact_offset[i] = 0;
+    }
+
+    return 0;
```

Comment:
yeah this is weird control flow, but I think something like this would be cleaner:

```
while (true) {
  read_entity
  all = true;
  for (...) if (fact_offset != fact_count) all = false;
  if (!all) break;
}

for (...) fact_offset = fact_count = 0;
```